src/app/common/services/gateways/gateways.service.ts

```diff
@@ -49,64 +50,91 @@ export class GatewaysService {
     return this.thingsService.addThing(gateway).map(
       resp => {
         const gwID = resp.headers.get('location').replace('/things/', '');
-
-        const ctrlChan: Channel = {
-          name: `${gateway.name}-${this.typeCtrlChan}`,
-          metadata: {
-            type: this.typeCtrlChan,
-          },
-        };
-        this.channelsService.addChannel(ctrlChan).subscribe(
-          respAddCtrl => {
-            const ctrlChanID = respAddCtrl.headers.get('location').replace('/channels/', '');
-
-            const dataChannel: Channel = {
-              name: `${gateway.name}-${this.typeDataChan}`,
+        this.thingsService.getThing(gwID).subscribe(
+          (respGetThing: any) => {
```

Comment:
Too long param name for the lambda. Just use `th` or `thing`. Instead of `any`, please think of using `Thing` interface from the interfaces file.

Reply:
Fixed `any` to `Thing`
This naming followed style for all other responses that was already in this file, like `respAddData`, `respConnectCtrl`, `respConnectData`...

Reply:
Alles klar.